pkg/operator/generic_client.go

```diff
@@ -3,64 +3,64 @@ package operator
 import (
 	"k8s.io/client-go/tools/cache"
 
-	operatorv1 "github.com/openshift/api/operator/v1"
-	clientv1 "github.com/openshift/cluster-openshift-controller-manager-operator/pkg/generated/clientset/versioned/typed/openshiftcontrollermanager/v1"
-	clientinformers "github.com/openshift/cluster-openshift-controller-manager-operator/pkg/generated/informers/externalversions"
+	operatorapiv1 "github.com/openshift/api/operator/v1"
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	operatorinformers "github.com/openshift/client-go/operator/informers/externalversions"
 )
 
 type operatorClient struct {
-	informers clientinformers.SharedInformerFactory
-	client    clientv1.OpenshiftcontrollermanagerV1Interface
+	informers operatorinformers.SharedInformerFactory
+	client    operatorclientv1.OperatorV1Interface
 }
 
 func (p *operatorClient) Informer() cache.SharedIndexInformer {
-	return p.informers.Openshiftcontrollermanager().V1().OpenShiftControllerManagerOperatorConfigs().Informer()
+	return p.informers.Operator().V1().OpenShiftControllerManagers().Informer()
 }
 
-func (p *operatorClient) CurrentStatus() (operatorv1.OperatorStatus, error) {
-	instance, err := p.informers.Openshiftcontrollermanager().V1().OpenShiftControllerManagerOperatorConfigs().Lister().Get("instance")
+func (p *operatorClient) CurrentStatus() (operatorapiv1.OperatorStatus, error) {
+	instance, err := p.informers.Operator().V1().OpenShiftControllerManagers().Lister().Get("instance")
 	if err != nil {
-		return operatorv1.OperatorStatus{}, err
+		return operatorapiv1.OperatorStatus{}, err
 	}
 
 	return instance.Status.OperatorStatus, nil
 }
 
-func (c *operatorClient) GetOperatorState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, string, error) {
-	instance, err := c.informers.Openshiftcontrollermanager().V1().OpenShiftControllerManagerOperatorConfigs().Lister().Get("instance")
+func (c *operatorClient) GetOperatorState() (*operatorapiv1.OperatorSpec, *operatorapiv1.OperatorStatus, string, error) {
+	instance, err := c.informers.Operator().V1().OpenShiftControllerManagers().Lister().Get("instance")
```

Comment:
possibly worth addressing in a separate PR - rename instance to `cluster`?

Reply:
i haven't kept up w/ what the other operators have done, but personally i like instance.

oc edit openshiftcontrollermanager.operator.openshift.io/instance makes more sense to me than oc edit openshiftcontrollermanager.operator.openshift.io/cluster

we used cluster for true "config-only" data, but in this case you're really driving an instance of something.  (the only instance of it that exists).